flink-libraries/flink-table/src/main/scala/org/apache/flink/table/calcite/RelTimeIndicatorConverter.scala

```diff
@@ -349,6 +350,14 @@ class RelTimeIndicatorConverter(rexBuilder: RexBuilder) extends RelShuttle {
             isTimeIndicatorType(updatedCall.getOperands.get(0).getType) =>
           updatedCall
 
+        case BasicOperatorTable.TUMBLE_ROWTIME |
+            BasicOperatorTable.TUMBLE_PROCTIME |
+            BasicOperatorTable.HOP_ROWTIME |
+            BasicOperatorTable.HOP_PROCTIME |
+            BasicOperatorTable.SESSION_ROWTIME |
+            BasicOperatorTable.SESSION_PROCTIME if isTimeIndicatorType(updatedCall.getType) =>
+          updatedCall
```

Comment:
I thought about this again, I think we have to use this approach. Because the operands are always materialized, because of the shape of the relational tree (groupings are always materialized which is correct).